src/main/java/de/kittybot/kittybot/cache/GuildSettingsCache.java

```diff
@@ -0,0 +1,164 @@
+		return getGuildSettings(guildId).getInactiveRoleId();
+	}
+
+	public static String getInactiveRole(String guildId){
+		return getGuildSettings(guildId).getInactiveRole();
+	}
+
+	public static void setCommandPrefix(String guildId, String prefix){
+		getGuildSettings(guildId).setCommandPrefix(prefix);
+		Database.setCommandPrefix(guildId, prefix);
+	}
+
+	public static void setRequestChannelId(String guildId, String channelId){
+		getGuildSettings(guildId).setRequestChannelId(channelId);
+		Database.setRequestChannelId(guildId, channelId);
+	}
+
+	public static void setRequestsEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setRequestsEnabled(enabled);
+		Database.setRequestsEnabled(guildId, enabled);
+	}
+
+	public static void setAnnouncementChannelId(String guildId, String channelId){
+		getGuildSettings(guildId).setAnnouncementChannelId(channelId);
+		Database.setAnnouncementChannelId(guildId, channelId);
+	}
+
+	public static void setJoinMessage(String guildId, String message){
+		getGuildSettings(guildId).setJoinMessage(message);
+		Database.setJoinMessage(guildId, message);
+	}
+
+	public static void setJoinMessagesEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setJoinMessagesEnabled(enabled);
+		Database.setJoinMessageEnabled(guildId, enabled);
+	}
+
+	public static void setLeaveMessage(String guildId, String message){
+		getGuildSettings(guildId).setLeaveMessage(message);
+		Database.setLeaveMessage(guildId, message);
+	}
+
+	public static void setLeaveMessagesEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setLeaveMessagesEnabled(enabled);
+		Database.setLeaveMessageEnabled(guildId, enabled);
+	}
+
+	public static void setBoostMessage(String guildId, String message){
+		getGuildSettings(guildId).setBoostMessage(message);
+		Database.setBoostMessage(guildId, message);
+	}
+
+	public static void setBoostMessagesEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setBoostMessagesEnabled(enabled);
+		Database.setBoostMessageEnabled(guildId, enabled);
+	}
+
+	public static void setLogChannelId(String guildId, String channelId){
+		getGuildSettings(guildId).setLogChannelId(channelId);
+		Database.setLogChannelId(guildId, channelId);
+	}
+
+	public static void setLogMessagesEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setLeaveMessagesEnabled(enabled);
+		Database.setLogMessagesEnabled(guildId, enabled);
+	}
+
+	public static void setNSFWEnabled(String guildId, boolean enabled){
+		getGuildSettings(guildId).setNSFWEnabled(enabled);
+		Database.setNSFWEnabled(guildId, enabled);
+	}
+
+	public static void setInactiveRoleId(String guildId, String roleId){
+		getGuildSettings(guildId).setInactiveRoleId(roleId);
+		Database.setInactiveRoleId(guildId, roleId);
+	}
+
+	public static void pruneCache(Guild guild){
+		SETTINGS.entrySet().removeIf(entry -> entry.getValue().getGuildId().equals(guild.getId()));
```

Comment:
why not just use `SETTINGS.remove(guild.getId())` here? i believe the map key is the guild id?

Reply:
Yes yes blondy copied that from my old branch lol